client/signup/steps/design-picker/index.jsx

```diff
@@ -74,20 +73,23 @@ export default function DesignPickerStep( props ) {
 	const {
 		isLoading: blockEditorSettingsAreLoading,
 		data: blockEditorSettings,
-	} = useBlockEditorSettingsQuery( siteId, userLoggedIn && ! props.useDIFMThemes );
+	} = useBlockEditorSettingsQuery( siteId, userLoggedIn && ! props.hideFSEThemes );
 	const isFSEEligible = blockEditorSettings?.is_fse_eligible ?? false;
-	const themeFilters = isFSEEligible
-		? 'auto-loading-homepage,full-site-editing'
-		: 'auto-loading-homepage';
+
+	const getThemeFilters = () => {
+		if ( props.themeFilters ) return props.themeFilters;
+
+		if ( isFSEEligible ) return 'auto-loading-homepage,full-site-editing';
+
+		return 'auto-loading-homepage';
+	};
 
 	const { data: apiThemes = [] } = useThemeDesignsQuery(
-		{ filter: themeFilters, tier },
+		{ filter: getThemeFilters(), tier },
 		// Wait until block editor settings have loaded to load themes
-		{ enabled: ! props.useDIFMThemes && ! blockEditorSettingsAreLoading }
+		{ enabled: props.hideFSEThemes || ! blockEditorSettingsAreLoading }
```

Comment:
I think we don't need `props.hideFSEThemes`. We could always wait for block editor settings to be ready and then fetch the themes.

Reply:
Doesn't it prevent an unnecessary network request in the DIFM flow though?

Reply:
No, DIFM hides FSE themes and always make network request after this change. I think this flag is try to request theme immediately and we don't need to wait for block editor settings to be ready

Reply:
Ooooh yes, I misunderstood. You're right we don't need `props.hideFSEThemes` here.

```suggestion
		{ enabled: ! blockEditorSettingsAreLoading }
```